terminate called after throwing an instance of 'std::length_error'   what():  vector::_M_range_insert

Hello everyone,
Thanks for this awesome project!

I am a complete newbie in Chromatography and I have been trying to view the results from a .wiff / .wiff.scan file acquired from a SCIEX QTrap device.

After converting the .wiff / .wiff.scan files to .mzML  using ProteoWizard ( https://proteowizard.sourceforge.io/ ) I get the .mzML files for the experiment and try to open them using TOPPView I get a crash after clicking on one of the peptides on the lower screen


![screen153](https://github.com/OpenMS/OpenMS/assets/97630/a0c37218-f71b-43bc-bbb2-b5f99faaa7f2)

This is the console log : 

```
./TOPPView 
Icon theme "elementary" not found.
Expecting a single source file in mzML. Found 2 will take only first one for rewriting.
..
INFO: done loading all 
terminate called after throwing an instance of 'std::length_error'
  what():  vector::_M_range_insert
Aborted
```

I have tried to both use the original file .mzML and also pass it through FileConverter to maybe strip compressed symbols  using : 

```
FileConverter -in  ~/Documents/Programming/multiquant/test-High\ A1.mzML -out ~/Documents/Programming/multiquant/testA1.mzML
```

I am using the latest master commit that in the about window of TOPPView is listed as 3.1.0-pre-develop-2023-11-21

I built the project on Ubuntu 22.04.3 LTS
on the following kernel 6.2.0-37-generic #38~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Nov  2 18:01:13 UTC 2 x86_64 x86_64 x86_64 GNU/Linux


Thank you for your help!

I am also uploading the two sample files that have the same behavior so you can reproduce my problem

[test.zip](https://github.com/OpenMS/OpenMS/files/13439623/test.zip)

Just an addition, I  noticed a warning about qtWebEngine so I also 
```
sudo apt install qtwebengine5-dev
```
and tried rebuilding the code using -DBOOST_USE_STATIC=On
However trying to build with static boost leads to the following build error.  

```
[ 25%] Linking CXX shared library ../../lib/libOpenMS.so
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libboost_regex.a(regex.o): warning: relocation against `_ZTVN5boost10wrapexceptISt13runtime_errorEE' in read-only section `.text.unlikely'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libboost_regex.a(instances.o): relocation R_X86_64_PC32 against symbol `_ZTVSt15basic_streambufIcSt11char_traitsIcEE@@GLIBCXX_3.4' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [src/openms/CMakeFiles/OpenMS.dir/build.make:11098: lib/libOpenMS.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:2252: src/openms/CMakeFiles/OpenMS.dir/all] Error 2
make: *** [Makefile:146: all] Error 2

```

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -23,6 +23,7 @@ Library:
 - better documentation for all SpectraFilter... tools (#7183)
 
 Fixes:
+- TOPPView: fix crash when viewing certain Chromatograms (#7220)
 - more robust parsing of mzIdentML (#7153)
 - more robust FileConverter (#7176)
 - MSFragger: allow relative path to database (#7155)

--- a/src/openms/source/KERNEL/MSExperiment.cpp
+++ b/src/openms/source/KERNEL/MSExperiment.cpp
@@ -287,8 +287,10 @@ namespace OpenMS
     // update intensity, m/z and RT according to chromatograms as well:
     for (ChromatogramType& cp : chromatograms_)
     {
+      // update range of EACH chrom, if we need them individually later
+      cp.updateRanges();
 
-      // ignore TICs and ECs (as these are usually positioned at 0 and therefor lead to a large white margin in plots if included)
+      // ignore TICs and ECs for the whole experiments range (as these are usually positioned at 0 and therefor lead to a large white margin in plots if included)
       if (cp.getChromatogramType() == ChromatogramSettings::TOTAL_ION_CURRENT_CHROMATOGRAM ||
         cp.getChromatogramType() == ChromatogramSettings::EMISSION_CHROMATOGRAM)
       {
@@ -299,7 +301,6 @@ namespace OpenMS
 
       // ranges
       this->extendMZ(cp.getMZ());// MZ
-      cp.updateRanges();
       this->extend(cp);// RT and intensity from chroms's range
     }
   }

--- a/src/openms_gui/include/OpenMS/VISUAL/LayerData1DChrom.h
+++ b/src/openms_gui/include/OpenMS/VISUAL/LayerData1DChrom.h
@@ -36,6 +36,11 @@ namespace OpenMS
 
     RangeAllType getRangeForArea(const RangeAllType partial_range) const override
     {
+      if (partial_range.RangeRT::isEmpty())
+      {
+        auto chrom = getCurrentChrom();
+        return RangeAllType().assign(chrom.getRange());
+      }
       const auto& chrom = getCurrentChrom();
       auto chrom_filtered = MSExperiment::ChromatogramType();
       chrom_filtered.insert(chrom_filtered.begin(), chrom.RTBegin(partial_range.getMinRT()), chrom.RTEnd(partial_range.getMaxRT()));

--- a/src/openms_gui/include/OpenMS/VISUAL/LayerData1DChrom.h
+++ b/src/openms_gui/include/OpenMS/VISUAL/LayerData1DChrom.h
@@ -36,10 +36,12 @@ namespace OpenMS
 
     RangeAllType getRangeForArea(const RangeAllType partial_range) const override
     {
+      // update ranges based on given RT range
       if (partial_range.RangeRT::isEmpty())
-      {
-        auto chrom = getCurrentChrom();
-        return RangeAllType().assign(chrom.getRange());
+      { // .. unless RT is empty, then we use the whole RT range
+        auto r = RangeAllType(partial_range);
+        r.extend(getCurrentChrom().getRange());
+        return r;
       }
       const auto& chrom = getCurrentChrom();
       auto chrom_filtered = MSExperiment::ChromatogramType();

